Makefile

```diff
@@ -1,61 +1,176 @@
+# -*- coding: utf-8; mode: makefile-gmake -*-
 # Basic Makefile
 
 UUID = system-monitor@paradoxxx.zero.gmail.com
-BASE_MODULES = $(UUID)/extension.js $(UUID)/README* $(UUID)/metadata.json $(UUID)/prefs.js $(UUID)/stylesheet.css $(UUID)/convenience.js $(UUID)/compat.js $(UUID)/gpu_usage.sh
-ifeq ($(strip $(DESTDIR)),)
-	INSTALLBASE = $(HOME)/.local/share/gnome-shell/extensions
+INSTALLNAME = $(UUID)
+DESTDIR ?= $(HOME)
+
+BASE_MODULES = \
+  $(UUID)/extension.js \
+  $(UUID)/README* \
+  $(UUID)/metadata.json \
+  $(UUID)/prefs.js \
+  $(UUID)/stylesheet.css \
+  $(UUID)/convenience.js \
+  $(UUID)/compat.js \
+  $(UUID)/gpu_usage.sh
+
+# ---------
+# variables
+# ---------
+
+ifeq ($(strip $(DESTDIR)),$(HOME))
+  INSTALLBASE = $(DESTDIR)/.local/share/gnome-shell/extensions
```

Comment:
To be honest, I'm not sure changing this from `$(HOME)` to `$(DESTDIR) ` (and defaulting `$DESTDIR` to `$HOME`) makes a lot of sense.

What's the use-case?

Reply:
From my POV `$DESTDIR` is where the installation goes (the base path, which might be root or home).  If the `$DESTDIR` is root also activate sudo. Or did I misunderstood you?

Reply:
Well, in the most technical sense, you could be considered correct.  However in practice `$DESTDIR` is usually a system-relative installation prefix for _system_ installations and not typically used for the `$HOME` type relocation, in my experience.

Additionally since this branch in the `Makefile` is absolutely predicated on _$DESTDIR == $HOME_ (i.e. it won't be entered under any other condition), it seems that resolving `$HOME` indirectly through `$DESTDIR` only adds confusion where it doesn't need to be.

Reply:
OK, thanks for clarifying, now  also read the [Makefile Conventions](https://www.gnu.org/prep/standards/html_node/DESTDIR.html) 

I can rollback the commits and do it like you showed in your first suggestion, but in your first  suggestion you also said

> This relies on the user realising/remembering to postfix / to their DESTDIR=. There is no harm in having multiple /s so leaving this one in causes no harm and is defensive.

If I rollback, there is no difference in using `make DESTDIR= install` and `make install` (DESTDIR is always empty), the installation always goes to home. What I mean, there is no need for an additional `/` in the `INSTALLBASE`, 

Anyway after reading [Makefile Conventions](https://www.gnu.org/prep/standards/html_node/DESTDIR.html)  I realized that I should not set the value of `DESTDIR` in Makefile at all. I will patch this tomorrow, today I'm to tired ... If you have any more hints for me in the meantime ..

_Hope I was understandable, my english is not the best and thanks for beeing patient with me._

Reply:
Right.  So my first suggestion was to just restore the `/` before the `usr/share/gnome-shell/extensions` like in the [original source](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/blob/56c4b5020e64c27cd037e6f3ff67d67619a6c628/Makefile#L8) so that users don't have to know to set `DESTDIR=/` if they don't want a _DESTDIR_.

Reply:
Hope to be clear, if `DESTDIR` is not set, it is empty and the installation goes into HOME (see [original source](https://github.com/paradoxxxzero/gnome-shell-system-monitor-applet/blob/56c4b5020e64c27cd037e6f3ff67d67619a6c628/Makefile#L5). Therefore an additional `/` in the `INSTALLBASE` is not necessary and that was exactly what disturbed me about the old implementation.  Anyway, I'll send a new series.